pkg/restapi/vc/operation/operations.go

```diff
@@ -731,40 +734,70 @@ func (o *Operation) createDIDUniRegistrar(keyType, signatureType, purpose string
 
 	identifier, keys, err := o.uniRegistrarClient.CreateDID(registrar.DriverURL, opts...)
 	if err != nil {
-		return "", "", "", fmt.Errorf("failed to create did doc from uni-registrar: %v", err)
+		return "", "", fmt.Errorf("failed to create did doc from uni-registrar: %v", err)
 	}
 
 	// TODO remove check when vendors supporting addKeys feature
 	if strings.Contains(identifier, "did:trustbloc") {
 		for _, v := range keys {
 			if strings.Contains(v.ID, "#"+selectedKeyID) {
-				return identifier, v.ID, "", nil
+				return identifier, v.ID, nil
 			}
 		}
 
-		return "", "", "", fmt.Errorf("selected key not found %s", selectedKeyID)
+		return "", "", fmt.Errorf("selected key not found %s", selectedKeyID)
 	}
 
 	if strings.Contains(identifier, "did:v1") {
 		for _, v := range keys {
 			for _, p := range v.Purpose {
 				if purpose == p {
-					return identifier, v.ID, v.PrivateKeyBase58, nil
+					err = o.importKey(v.ID, kms.ED25519Type, base58.Decode(v.PrivateKeyBase58))
+					if err != nil {
+						return "", "", err
+					}
+
+					return identifier, v.ID, nil
 				}
 			}
 		}
 
-		return "", "", "", fmt.Errorf("did:v1 - not able to find key with purpose %s", purpose)
+		return "", "", fmt.Errorf("did:v1 - not able to find key with purpose %s", purpose)
 	}
 
 	// vendors not supporting addKeys feature.
 	// return first key public and private
 	// TODO remove when vendors supporting addKeys feature
-	return identifier, keys[0].ID, keys[0].PrivateKeyBase58, nil
+	err = o.importKey(keys[0].ID, kms.ED25519Type, base58.Decode(keys[0].PrivateKeyBase58))
+	if err != nil {
+		return "", "", err
+	}
+
+	return identifier, keys[0].ID, nil
+}
+
+func (o *Operation) importKey(keyID string, keyType kms.KeyType, privateKeyBytes []byte) error {
+	split := strings.Split(keyID, "#")
+
+	var privKey interface{}
+
+	switch keyType {
+	case kms.ED25519Type:
```

Comment:
Is import only supported for Ed25519 for now or forever?

Reply:
for now